core/store/models/p2pkey/p2p_key.go

```diff
@@ -0,0 +1,123 @@
+package p2pkey
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	keystore "github.com/ethereum/go-ethereum/accounts/keystore"
+	cryptop2p "github.com/libp2p/go-libp2p-core/crypto"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
+)
+
+// Key represents a libp2p private key
+type Key struct {
+	cryptop2p.PrivKey
+}
+
+func (k Key) GetPeerID() (peer.ID, error) {
+	peerID, err := peer.IDFromPrivateKey(k)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return peerID, err
+}
+
+type EncryptedP2PKey struct {
+	ID               int32 `gorm:"primary_key"`
+	PeerID           string
+	PubKey           []byte
+	EncryptedPrivKey []byte
+	CreatedAt        time.Time
+	UpdatedAt        time.Time
+}
+
+func (EncryptedP2PKey) TableName() string {
+	return "encrypted_p2p_keys"
+}
+
+// CreateKey makes a new libp2p keypair from a crytographically secure entropy source
+func CreateKey() (Key, error) {
+	p2pPrivkey, _, err := cryptop2p.GenerateEd25519Key(rand.Reader)
+	if err != nil {
+		return Key{}, nil
+	}
+	return Key{
+		p2pPrivkey,
+	}, nil
+}
+
+type ScryptParams struct{ N, P int }
+
+var defaultScryptParams = ScryptParams{
+	N: keystore.StandardScryptN, P: keystore.StandardScryptP}
+
+// type is added to the beginning of the passwords for
+// P2P key, so that the keys can't accidentally be mis-used
+// in the wrong place
+func adulteratedPassword(auth string) string {
+	s := "p2pkey" + auth
+	return s
+}
+
+func (k Key) Encrypt(auth string, p ...ScryptParams) (s EncryptedP2PKey, err error) {
```

Comment:
I don't understand why this method needs to have this signature? 
I can't find any call of Encrypt with params other than the password?
Is there an interface that `Key` needs to implement? If so, can you add a comment on this method with that interface's name.